resources/splunk_app.rb

```diff
@@ -69,25 +65,43 @@ def parse_meta_access(access)
     end
   end
 
+  # Provider exclusive methods
+
   action_class do
     include CernerSplunk::ProviderHelpers
 
+    def upgrade_app(strategy)
+      require 'fileutils'
+
+      case strategy
+      when :prep
+        app_cache_path.mkpath
+        FileUtils.mv(app_path, app_cache_path + name)
+      when :keep_existing
+        FileUtils.cp_r(app_cache_path + name + 'local', app_path + 'local')
+        CernerSplunk::FileHelpers.deep_change_ownership(app_path + 'local', current_owner, current_group)
+
+        FileUtils.cp(app_cache_path + name + 'metadata/local.meta', app_path + 'metadata/local.meta')
```

Comment:
We want to prefer the apps lookups but that's not a universal truth we need to give the choice on it. But also restoring the local data without restoring the lookup files, there could be transforms stanzas and permissions that would refer to a lookup that was built through the UI, that would stop functioning with this